24_clustering.Rmd

```diff
@@ -221,97 +196,122 @@ tse <- GlobalPatterns
+clusters <- as.data.frame(cbind(clusters, levels(colData(tse_dmm)$SampleType)[colData(tse_dmm)$SampleType])) # add sample type information
+colnames(clusters) <- c("Group1", "Group2", "SampleType")
 
-```{r dmm7}
-DirichletMultinomial::mixturewt(bestFit)
+clusters %>%
+  group_by(SampleType) %>%
+  arrange(Group1) %>%
+  filter(row_number()==1)
 ```
 
-It's also possible to get the samples-cluster assignment probabilities: how
-probable it is that each sample belongs to each cluster
+We can also plot the driver Phyla in each group. In this case, it reflects the differences between environmental and human samples. 
 
-```{r dmm8}
-prob <- metadata(tse)$DMM$prob
-head(prob)
+```{r dmm9, fig.width=8, fig.height=4}
+# Get the estimates on how much each phyla contributes on each cluster
+best_model <- metadata(tse_dmm)$DMM$dmm[2]
+drivers <- as.data.frame(best_model[[1]]@fit$Estimate)
+
+drivers$phyla <- gsub("Phylum:", "", rownames(drivers)) # Clean phylum names
+
+plots <- c()
+for (i in 1:2) {
+  drivers <- drivers[order(drivers[[i]], decreasing = TRUE),]
+  p <- ggplot(head(drivers, 10), aes(x = reorder(head(phyla, 10), + head(drivers[[i]], 10)), y = head(drivers[[i]], 10))) +
+          geom_bar(stat = "identity", fill = "deeppink4", alpha = 0.5) +
+          coord_flip() + labs(title = paste("Top phyla in a group", i)) +
+          theme_light(base_size = 15) + xlab("") + ylab("") + ylim(0, 7)
+  print(p)
+}
 ```
 
-We can also know the contribution of each taxa to each component
 
-```{r dmm9}
-head(DirichletMultinomial::fitted(bestFit))
+We use `calculateDMNgroup` function to have an overview of the best model. The function groups samples by sample type and returns DMNGroup object that contains a summary.
+
+```{r dmm_group}
+dmm_group <- calculateDMNgroup(tse_dmm, variable = "SampleType", 
+                               assay.type = "counts", k = 2, 
+                               seed = .Machine$integer.max)
+
+dmm_group
 ```
 
-Finally, to be able to visualize our data and clusters, we start by 
-computing the euclidean PCoA and storing it as a data frame.
+Mixture weights can be used for having a rough approximation of the cluster size.
 
 ```{r dmm10}
-# Do clr transformation. Pseudocount is added, because data contains zeros.
+DirichletMultinomial::mixturewt(bestFit)
+```
+
+### PCoA with DMM clusters
+
+In this section we show how to calculate principal coordinates for clr transformed abundance data. We use Aitchison distance as a distance metrics that calculates Euclidean distances for clr transformed compositions. In the visualization section, the samples are coloured based on their DMM clusters to demonstrate the distances between samples included in each DMM cluster.
+
+```{r dmm11}
+# pseudocount is added, because data contains zeros
 assay(tse, "pseudo") <- assay(tse, "counts") + 1
 tse <- transformAssay(tse, assay.type = "pseudo", method = "relabundance")
+
+# clr transformation
```

Comment:
It seems that data transformations are done many times in this chapter. Could we just add the transformed assays once in the beginning of the chapter, then call these by assay name in the later chunks?

Reply:
Hclust and DMM examples all use different data all saved with the same name "tse". Should I change it (either names to be different or datas to be the same)? I also made some editions to transformations

Reply:
Let us use just one name "tse" unless there is specific need to use many at the same time. 

Using just "tse" makes it way easier to copypaste and reuse the code for  new purposes.